### PR TITLE
Fix application crash in tt-rss API from unreads

### LINKED ISSUE #691: Application crashes on first sync attempt with tt-rss
Version:
```
Version: 2.4-dev
Git Commit: 7de9c3238fe7c5cb39c8fe4c5917ef88b3c9c79e
```

Stack trace:
```
An error occured (SIGSEGV) in ../../../tmp/FeedReader/plugins/backend/ttrss/ttrssAPI.vala, line 194 in 'feed_reader_ttrss_api_getUnreadCount'
   The reason is likely a null reference being used.

    #1  <unknown>                                                                 in 'atoi'
        at /lib/x86_64-linux-gnu/libc.so.6
 *  #2  ../../../tmp/FeedReader/plugins/backend/ttrss/ttrssAPI.vala    line  194 in 'feed_reader_ttrss_api_getUnreadCount'
        at /tmp/FeedReader/plugins/backend/ttrss/ttrssAPI.vala:194
    #3  ../../../tmp/FeedReader/build/FeedServerInterface.c            line  841 in 'feed_reader_feed_server_interface_getUnreadCount'
        at /tmp/FeedReader/build/FeedServerInterface.c:841
    #4  ../../../tmp/FeedReader/src/Backend/FeedServer.vala            line 1002 in 'feed_reader_feed_server_getUnreadCount'
        at /tmp/FeedReader/src/Backend/FeedServer.vala:1002
    #5  ../../../tmp/FeedReader/src/Backend/FeedServer.vala            line  239 in 'feed_reader_feed_server_syncContent'
        at /tmp/FeedReader/src/Backend/FeedServer.vala:239
```

[L194](https://github.com/jangernert/FeedReader/blob/master/plugins/backend/ttrss/ttrssAPI.vala#L194). This is my first time looking at Vala. Is `response` null?

### PR DESCRIPTION
**Pull request by allanbreyes**
Fixes #691. This is the first time I've touched Vala, so hopefully this makes sense.

References:
- https://stackoverflow.com/questions/20777923/vala-int-parse-crashing
- https://valadoc.org/glib-2.0/int64.try_parse.html

### CODE REVIEW

**Comment by allanbreyes** [comment]
Note: I don't know if it's related, but after implementing this, I still ran into #692.

File: plugins/backend/ttrss/ttrssAPI.vala [hunk 1]
```diff
@@ -191,7 +191,15 @@ public class FeedReader.ttrssAPI : GLib.Object {
...
 			var response = message.get_response_object();
-			unread = int.parse(response.get_string_member("unread"));
+			int64 parsed = 0;
+			bool res = int64.try_parse(response.get_string_member("unread"), out parsed);
+			if(res) {
+				unread = (int) parsed;
+			}
+			else
+			{
+				Logger.warning("Could not parse unread articles");
```
Comment (brendanlong) [inline plugins/backend/ttrss/ttrssAPI.vala]: I wonder if this should be `error` level?

Comment (allanbreyes) [reply plugins/backend/ttrss/ttrssAPI.vala]: I went with the web dev mantra of, "Should this wake up a sysadmin at night?" Yes: `error`, No: `warning`. Didn't seem severe enough that it warranted an error log level...

**Comment by allanbreyes** [comment]
FWIW:

```
$ grep -R int.parse | wc -l
48
```

Possibly a red herring for #692, but I see lots of date conversions... `int.parse` might be the culprit for those. :thinking:

**Pull request merged**